src/connectors/dynamic-widgets/__tests__/connectDynamicWidgets-test.ts

```diff
@@ -0,0 +1,581 @@
+            Widget(ais.dynamicWidgets),
+          ]
+        `);
+
+        dynamicWidgets.init!(createInitOptions({ parent }));
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
+          Array [
+            Widget(ais.dynamicWidgets),
+            Widget(ais.menu) {
+              attribute: test1
+            },
+            Widget(ais.hierarchicalMenu) {
+              attribute: test2
+            },
+          ]
+        `);
+
+        dynamicWidgets.render!(createRenderOptions({ parent }));
+
+        await runAllMacroTasks();
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
+          Array [
+            Widget(ais.dynamicWidgets),
+            Widget(ais.menu) {
+              attribute: test1
+            },
+          ]
+        `);
+      });
+
+      it('renders widgets returned by transformItems', async () => {
+        const dynamicWidgets = EXPERIMENTAL_connectDynamicWidgets(() => {})({
+          transformItems(_items, { results }) {
+            return results.userData[0].MOCK_facetOrder;
+          },
+          widgets: [
+            connectMenu(() => {})({ attribute: 'test1' }),
+            connectHierarchicalMenu(() => {})({
+              attributes: ['test2', 'test3'],
+            }),
+          ],
+        });
+
+        const parent = index({ indexName: 'test' }).addWidgets([
+          dynamicWidgets,
+        ]);
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
+          Array [
+            Widget(ais.dynamicWidgets),
+          ]
+        `);
+
+        dynamicWidgets.init!(createInitOptions({ parent }));
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
+          Array [
+            Widget(ais.dynamicWidgets),
+            Widget(ais.menu) {
+              attribute: test1
+            },
+            Widget(ais.hierarchicalMenu) {
+              attribute: test2
+            },
+          ]
+        `);
+
+        dynamicWidgets.render!(
+          createRenderOptions({
+            parent,
+            results: new SearchResults(
+              new SearchParameters(),
+              createMultiSearchResponse({
+                userData: [{ MOCK_facetOrder: ['test1'] }],
+              }).results
+            ),
+          })
+        );
+
+        await runAllMacroTasks();
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
+          Array [
+            Widget(ais.dynamicWidgets),
+            Widget(ais.menu) {
+              attribute: test1
+            },
+          ]
+        `);
+
+        dynamicWidgets.render!(
+          createRenderOptions({
+            parent,
+            results: new SearchResults(
+              new SearchParameters(),
+              createMultiSearchResponse({
+                userData: [{ MOCK_facetOrder: ['test2', 'test1'] }],
+              }).results
+            ),
+          })
+        );
+
+        await runAllMacroTasks();
+
+        expect(parent.getWidgets()).toMatchInlineSnapshot(`
```

Comment:
why isn't the order test2 → test1 ?

Reply:
the widgets themselves don't have a useful concept of ordering, therefore the only important thing is that the right ones are mounted more than their order.